File: app/src/main/java/com/kidozh/discuzhub/results/SecureInfoResult.java
package com.kidozh.discuzhub.results;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class SecureInfoResult extends BaseResult {
    @JsonProperty("Variables")
    public SecureVariables secureVariables;

    public static class SecureVariables extends VariableResults{

        @JsonProperty("sechash")
        public String secHash;
        @JsonProperty("seccode")
        public String secCodeURL;

    }

    public boolean isError(){
        return this.message == null;
    }
}

File: app/src/main/java/com/kidozh/discuzhub/results/SecureInfoResult.kt
package com.kidozh.discuzhub.results

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class SecureInfoResult : BaseResult() {
    @JvmField
    @JsonProperty("Variables")
    var secureVariables: SecureVariables? = null

    class SecureVariables : VariableResults() {
        @JvmField
        @JsonProperty("sechash")
        var secHash: String = ""

        @JvmField
        @JsonProperty("seccode")
        var secCodeURL: String = ""
    }

    override fun isError(): Boolean {
        return message == null
    }
}